ai2thor/server.py

```diff
@@ -85,15 +85,39 @@ def unique_rows(arr, return_index=False, return_inverse=False):
         return unique
 
 
-class Event(object):
+class MetadataWrapper(dict):
+    def __init__(self, raw_metadata: dict):
+        super().__init__(raw_metadata)
+        self._raw_metadata = raw_metadata
+
+    def __getitem__(self, x):
+        # alias deprecated functionality
+        if (
+            x == "reachablePositions"
+            and self._raw_metadata["lastAction"] == "GetReachablePositions"
+        ):
+            warnings.warn(
+                'The key event.metadata["reachablePositions"] is deprecated and has been remapped to event.metadata["actionReturn"].'
+            )
+            x = "actionReturn"
+        elif x == "reachablePositions":
+            raise IndexError(
+                "You are trying to access event.metadata['reachablePositions'] without first "
+                + "calling controller.step(action='GetReachablePositions'). Also, "
+                + "the key 'reachablePositions' is deprecated in favor of event.metadata['actionReturn']."
+            )
```

Comment:
Note: I am intentionally throwing an error if they try to access `reachablePositions` without first calling `action="GetReachablePositions"`.